x-pack/plugins/enterprise_search/public/applications/applications/components/engine/engine_connect/engine_api_integration.tsx

```diff
@@ -13,52 +13,58 @@ import { compressToEncodedURIComponent } from 'lz-string';
 import {
   EuiCodeBlock,
   EuiFlexGroup,
+  EuiFlexItem,
   EuiLink,
   EuiSpacer,
   EuiTab,
   EuiTabs,
   EuiText,
 } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
+
 import { FormattedMessage } from '@kbn/i18n-react';
 
 import { useCloudDetails } from '../../../../shared/cloud_details/cloud_details';
+import { docLinks } from '../../../../shared/doc_links';
 import { KibanaLogic } from '../../../../shared/kibana';
 import { EngineViewLogic } from '../engine_view_logic';
 
 import { EngineApiLogic } from './engine_api_logic';
 
 import { elasticsearchUrl } from './search_application_api';
 
-const SearchUISnippet = (esUrl: string, engineName: string, apiKey: string) => `
-import EnginesAPIConnector from "@elastic/search-ui-engines-connector";
+const clientSnippet = (esUrl: string, searchApplicationName: string, apiKey: string) => `
+import Client from '@elastic/search-application-client'
+// or through CDN
+// const Client = window['SearchApplicationClient']
 
-const connector = new EnginesAPIConnector({
-  host: "${esUrl}",
-  engineName: "${engineName}",
-  apiKey: "${apiKey || '<YOUR_API_KEY>'}"
-});`;
+const request = Client(
+  ${searchApplicationName},
+  ${esUrl},
+  ${apiKey || '<YOUR_API_KEY>'},
+)
 
-const cURLSnippet = (esUrl: string, engineName: string, apiKey: string, params: unknown) => `
-curl --location --request POST '${esUrl}/_application/search_application/${engineName}/_search' \\
---header 'Authorization: apiKey ${apiKey || '<YOUR_API_KEY>'}' \\
---header 'Content-Type: application/json' \\
---data-raw '${JSON.stringify({ params }, null, 2)}'`;
+const results = await request()
+  .query('pizza')
+  .addParameter('myCustomParameter', 'example value')
+  .search()
+`;
 
-const apiRequestSnippet = (
+const cURLSnippet = (
   esUrl: string,
   searchApplicationName: string,
   apiKey: string,
   params: unknown
-) => {
+) => `
+curl --location --request POST '${esUrl}/_application/search_application/${searchApplicationName}/_search' \\
+--header 'Authorization: apiKey ${apiKey || '<YOUR_API_KEY>'}' \\
+--header 'Content-Type: application/json' \\
+--data-raw '${JSON.stringify({ params }, null, 2)}'`;
+
+const apiRequestSnippet = (searchApplicationName: string, params: unknown) => {
   const body = JSON.stringify({ params }, null, 2);
   return `
 POST /_application/search_application/${searchApplicationName}/_search HTTP/1.1
```

Comment:
remove the HTTP/1.1

Reply:
Added in e0c51f66718e079c83271b9d42edd93df97d73f1